python/ml4ir/applications/ranking/tests/test_metrics.py

```diff
@@ -88,6 +151,35 @@ def test_model_training(self):
             assert gold_metric_name in metrics
             assert np.isclose(metrics[gold_metric_name], gold_metric_val, atol=0.05)
 
+    def test_stat_sig_evaluation(self):
+        """testing ml4ir stat sig computation end-to-end"""
+
+        ROOT_DATA_DIR = "ml4ir/applications/ranking/tests/data"
+        feature_config_path = os.path.join(
+            ROOT_DATA_DIR, "configs", "feature_config_aux_loss.yaml"
+        )
+        model_config_path = os.path.join(
+            ROOT_DATA_DIR, "configs", "model_config_cyclic_lr.yaml")
+        eval_config_path = os.path.join(
+            ROOT_DATA_DIR, "configs", "aux_metrics_evaluation_config.yaml")
+
+        data_dir = os.path.join(ROOT_DATA_DIR, "tfrecord")
+        aux_loss = "aux_one_hot_cross_entropy"
+        self.train_ml4ir(data_dir, feature_config_path,
+                    model_config_path, eval_config_path, self.log_dir, aux_loss)
+
+        results_dict = pd.read_csv(
+            os.path.join(self.log_dir, "test_aux_loss", "_SUCCESS"), header=None
+        ).set_index(0).to_dict()[1]
+
+        expected_metrics = {
+            'stat_sig_MRR_improved_groups': '0', 'stat_sig_MRR_degraded_groups': '5', 'stat_sig_MRR_group_improv_perc': '-14.688059131571537',
+            'stat_sig_AuxIntrinsicFailure_improved_groups': '5', 'stat_sig_AuxIntrinsicFailure_degraded_groups': '0',
+            'stat_sig_AuxIntrinsicFailure_group_improv_perc': '98.01304093',
+            'stat_sig_AuxRankMF_improved_groups': '0', 'stat_sig_AuxRankMF_degraded_groups': '0' ,'test_perc_improv_AuxRankMF': '55.679052825410714'
```

Comment:
@mohazahran This feels like the wrong key - `'test_perc_improv_AuxRankMF'`
 Did you mean to add `stat_sid_AuxRank_group_improv_perc`?

Also, maybe we can call it `..._perc_improv_macro`?

Reply:
Added